### ISSUE TITLE
.deb installer for Buster or .appimage that works when extracted in / opt /

### ISSUE THREAD

**Issue by quirinux-so** [turtletooth/tahoma2d#384]
_This would not actually be an issue, but ..._

As I have not been able to compile Tahoma in Buster (ref. https://github.com/turtletooth/tahoma2d/issues/383 ) I am interested in extracting the content of the .appimage in order to execute it from a subdirectory of / opt /

This is something that I usually do with other programs in .appimage, since I noticed how the performance of the extracted appimages improves considerably with several programs to which I could apply this method.

However, although the files are extracted correctly with **--appimage-extract** then AppRun does not execute:

**QStandardPaths: XDG_RUNTIME_DIR not set, defaults to '/ tmp / runtime-root'
varName: Tahoma2DPROFILES TOONZROOT not set ...
varName: Tahoma2DPROFILES TOONZROOT not set ...
AppRun: /home/johndancel/OpenToonz-git/tahoma2d/toonz/sources/toonzlib/preferences.cpp:245: Preferences :: Preferences (): The `! M_roomMaps.isEmpty () 'is not satisfied.
Aborted**

Will there be a possibility to generate the next .appimage in such a way that they can then be extracted and executed that way? My main objective is to redistribute Tahoma within the Quirinux distribution and the ideal would be to be able to compile it **_or, better, to have a .deb..._** ;).

**Comment by manongjohn** [comment]
This would be true if you were able to compile or extract the components out of the AppImage.

Tahoma2D currently expects that the `tahomastuff` folder is in the same directory as where you are running the Tahoma2D executable from or passed in a configuration file telling it where to find it.  Not 100% sure passing the configuration file will work at the moment.

Although you are extracting the contents of the AppImage, you are likely running it away from where the tahomastuff folder is.

**Comment by quirinux-so** [comment]
Thank you very much for your patience. I tried putting tahomastuff inside:

- /opt/Tahoma2D
- /opt/Tahoma2D/usr/
- /opt/Tahoma2D/usr/bin

However I had no luck :-(

```
/opt/Tahoma2D$ ls

AppRun  ffmpeg  io.github.Tahoma2D.desktop  io.github.Tahoma2D.png  tahomastuff  usr

/opt/Tahoma2D$ ./AppRun 

plugin search directory:/opt/Tahoma2D/usr/bin/tahomastuff/plugins
walkDirectory_: /opt/Tahoma2D/usr/bin/tahomastuff/plugins
===== PluginLoadController::finished() =====
QLayout: Attempting to add QLayout "" to MainWindow "", which already has a layout
QLayout: Attempting to add QLayout "" to MainWindow "", which already has a layout
QLayout: Attempting to add QLayout "" to AboutPopup "", which already has a layout
QLayout: Attempting to add QLayout "" to TPanelTitleBarButtonForGrids "", which already has a layout
QLayout: Attempting to add QLayout "" to TPanelTitleBarButtonForGrids "", which already has a layout
QLayout: Attempting to add QLayout "" to TPanelTitleBarButtonForGrids "", which already has a layout
QLayout: Attempting to add QLayout "" to TPanelTitleBarButtonForGrids "", which already has a layout
QLayout: Attempting to add QLayout "" to TPanelTitleBarButtonForGrids "", which already has a layout
QLayout: Attempting to add QLayout "" to TPanelTitleBarButtonForGrids "", which already has a layout
QLayout: Attempting to add QLayout "" to TPanelTitleBarButtonForGrids "", which already has a layout
QLayout: Attempting to add QLayout "" to TPanelTitleBarButtonForGrids "", which already has a layout
QLayout: Attempting to add QLayout "" to TPanelTitleBarButtonForGrids "", which already has a layout
QLayout: Attempting to add QLayout "" to TPanelTitleBarButtonForGrids "", which already has a layout
QLayout: Attempting to add QLayout "" to TPanelTitleBarButtonForGrids "", which already has a layout
QLayout: Attempting to add QLayout "" to TPanelTitleBarButtonForGrids "", which already has a layout
QLayout: Attempting to add QLayout "" to TPanelTitleBarButtonForGrids "", which already has a layout
QLayout: Attempting to add QLayout "" to TPanelTitleBarButtonForGrids "", which already has a layout
QLayout: Attempting to add QLayout "" to TPanelTitleBarButtonForGrids "", which already has a layout
QLayout: Attempting to add QLayout "" to TPanelTitleBarButtonForGrids "", which already has a layout
QLayout: Attempting to add QLayout "" to TPanelTitleBarButtonForGrids "", which already has a layout
QLayout: Attempting to add QLayout "" to TPanelTitleBarButtonForGrids "", which already has a layout
QLayout: Attempting to add QLayout "" to TPanelTitleBarButtonForGrids "", which already has a layout
QLayout: Attempting to add QLayout "" to TPanelTitleBarButtonForGrids "", which already has a layout
plugin search directory:/opt/Tahoma2D/usr/bin/tahomastuff/plugins
walkDictionary: /opt/Tahoma2D/usr/bin/tahomastuff/plugins [dry]
QFileSystemWatcher::addPath: path is empty
inotify_add_watch("/opt/Tahoma2D/usr/bin/tahomastuff/sandbox") failed: "No such file or directory"
terminate called after throwing an instance of 'TSystemException'
Aborted

/opt/Tahoma2D$ cd /opt/Tahoma2D/usr/bin

/opt/Tahoma2D/usr/bin$ ./Tahoma2D 

plugin search directory:/opt/Tahoma2D/usr/bin/tahomastuff/plugins
walkDirectory_: /opt/Tahoma2D/usr/bin/tahomastuff/plugins
===== PluginLoadController::finished() =====
QLayout: Attempting to add QLayout "" to MainWindow "", which already has a layout
QLayout: Attempting to add QLayout "" to MainWindow "", which already has a layout
QLayout: Attempting to add QLayout "" to AboutPopup "", which already has a layout
QLayout: Attempting to add QLayout "" to TPanelTitleBarButtonForGrids "", which already has a layout
QLayout: Attempting to add QLayout "" to TPanelTitleBarButtonForGrids "", which already has a layout
QLayout: Attempting to add QLayout "" to TPanelTitleBarButtonForGrids "", which already has a layout
QLayout: Attempting to add QLayout "" to TPanelTitleBarButtonForGrids "", which already has a layout
QLayout: Attempting to add QLayout "" to TPanelTitleBarButtonForGrids "", which already has a layout
QLayout: Attempting to add QLayout "" to TPanelTitleBarButtonForGrids "", which already has a layout
QLayout: Attempting to add QLayout "" to TPanelTitleBarButtonForGrids "", which already has a layout
QLayout: Attempting to add QLayout "" to TPanelTitleBarButtonForGrids "", which already has a layout
QLayout: Attempting to add QLayout "" to TPanelTitleBarButtonForGrids "", which already has a layout
QLayout: Attempting to add QLayout "" to TPanelTitleBarButtonForGrids "", which already has a layout
QLayout: Attempting to add QLayout "" to TPanelTitleBarButtonForGrids "", which already has a layout
QLayout: Attempting to add QLayout "" to TPanelTitleBarButtonForGrids "", which already has a layout
QLayout: Attempting to add QLayout "" to TPanelTitleBarButtonForGrids "", which already has a layout
QLayout: Attempting to add QLayout "" to TPanelTitleBarButtonForGrids "", which already has a layout
QLayout: Attempting to add QLayout "" to TPanelTitleBarButtonForGrids "", which already has a layout
QLayout: Attempting to add QLayout "" to TPanelTitleBarButtonForGrids "", which already has a layout
QLayout: Attempting to add QLayout "" to TPanelTitleBarButtonForGrids "", which already has a layout
QLayout: Attempting to add QLayout "" to TPanelTitleBarButtonForGrids "", which already has a layout
QLayout: Attempting to add QLayout "" to TPanelTitleBarButtonForGrids "", which already has a layout
QLayout: Attempting to add QLayout "" to TPanelTitleBarButtonForGrids "", which already has a layout
plugin search directory:/opt/Tahoma2D/usr/bin/tahomastuff/plugins
walkDictionary: /opt/Tahoma2D/usr/bin/tahomastuff/plugins [dry]
QFileSystemWatcher::addPath: path is empty
inotify_add_watch("/opt/Tahoma2D/usr/bin/tahomastuff/sandbox") failed: "No such file or directory"
terminate called after throwing an instance of 'TSystemException'
Aborted

/opt/Tahoma2D/usr$ ls
bin  ffmpeg  lib  plugins  share  tahomastuff


```

**Comment by quirinux-so** [comment]
Hi:

I found out what my mistake was: I had to save the tahomastuff folder inside Tahoma2D/usr/bin.

I close this thread commenting that it is final: Tahoma2D will be included in Quirinux GNU/Linux 2.0 (Thank you very much!) And I am very happy for the release released yesterday.

A greeting.
